sheets/2/exercise-6.tex

```diff
@@ -0,0 +1,74 @@
+    \item Instanziiere den Task und den Verifier. Dabei soll wie in der \textit{Aufgabe 5b)} der SubTask jeweils angegeben werden.
+    \item Lasse Neo im Uhrzeigersinn an der Wand entlang laufen, bis er wieder an seinem Startpunkt angekommen ist.
+    Implementiere das Verhalten in der Methode \textit{subTaskB()} in der Klasse \textit{Sheet3Task6}.
+    \item Lasse nun Neo gegen den Uhrzeigersinn an der Wand entlang laufen, bis er wieder an seinem Startpunkt angekommen ist.
+    Implementiere das Verhalten in der Methode \textit{subTaskC()} in der Klasse \textit{Sheet3Task6}.
+    Was fällt dir im Vergleich zu vorhin auf?
+\end{enumerate}
+\begin{Infobox}
+    Wir wollen uns nun die Methodensyntex etwas genauer anschauen. Dafür betrachten wir beispielhaft den folgenden Programmcode:
+    \begin{lstlisting}[breaklines=true, numbers=none]
+//Class A
+public void foo(){
+    B b = new B();
+    System.out.println(b.bar(5, 7));
+}
+
+//Class B
+private int b = 9;
+public boolean bar(int a, int b){
+    if(a > b)
+        return true;
+    return power(a) > this.b;
+}
+
+private int power(int a){
+    return a*a;
+}
+\end{lstlisting}
+Bei der Programmausführung gehen wir davon aus dass die Methode \textit{foo} ausgeführt wird. Wir unterteilen die Erklärung der Syntax in folgende Blöcke:
+\begin{enumerate}[label=\roman*)]
+    \item 
+    Grundlegender Sytaxaufbau: Der Aufbau einer \textit{einfachen} Methode folgt zunächst einmal dem Schema 
+    \begin{lstlisting}
+Modifier Datentyp Methodenname (Par a, Par b, ...){
+    //do stuff
+    return ...;
+}
+    \end{lstlisting}
+    Welche Methoden sind also in dem obigen Codebeispiel gegeben und wie wurden diese benannt?
+    \item Modifier: Hier unterscheiden wir zunächst zwischen den \textit{public} und \textit{private} Modifiern. Der \textit{public} Modifier erlaubt es, dass die Methode aus einer
+    anderen beliebigen Klasse aufgerufen werden kann. Im Gegensatz dazu kann eine Methode mit dem \textit{private} Modifier nur innerhalb der selben Klasse aufgerufen werden.
+    Somit ist es beispielsweise möglich, dass die Methode \textit{foo} in der Klasse A die Methode \textit{bar} von der Klasse B aufruft, während Klasse A jedoch die \textit{power} Methode
+    der Klasse B nicht sehen kann.
+    \item Datentyp: Dieser gibt an, welche Art von Rückgabe die Methode liefert. Die Rückgabe selber wird durch das Keyword \textit{return} signalisiert.
+    \item \textit{return}:
+    Nachdem \textit{return} aufgerufen wurde, wird sofort zu der aufrufenden Methode zurückgesprungen.\\
+    Beispiel: Wenn die Methode \textit{bar} mit Parametern \textit{a = 5, b = 2} aufgerufen wird, ist die IF-Condition erfüllt und es wird sofort \textit{true} zurückgegeben.
+    Dadurch wird die darunterliegende Zeile nicht ausgeführt.
+    \item \textit{this}: Das \textit{this} Keyword erlaubt ist eine Referenz auf das aktuelle Objekt. Im Bezug auf die Felder der Klasse und auf das obige Code Beispiel erlaubt uns das 
```

Comment:
Referenzen wollen wir nach möglichkeit gar nicht erklären. Da muss man dann kreative erklärungen finden die das Wort auch umgehen. 

Vielleicht sowas in der Art:
"this" kann wie eine Variable benutzt werden. Der Wert von this ist das aktuelle Objekt von der Klasse in der sich die Funktion befindet. Das this in Neo::move() ist also ein Neo objekt. Und wenn man player.move() aufruft ist das this in Neo::move() (weil player ein Neo ist) gleich dem Wert von Player.